### PR TITLE
Aad 276 create push notification function app

### PR DESCRIPTION
**Pull request by CathyGroom**
Merge required with main.

### CODE REVIEW

File: src/SFA.DAS.PushNotifications.Functions/SFA.DAS.PushNotifications.Data/Repositories/ApplicationClientRepository.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private readonly IPushNotificationsDataContext _context;
+
+    public ApplicationClientRepository(IPushNotificationsDataContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<int> AddWebPushNotificationSubscription(ApplicationClient applicationClient, CancellationToken cancellationToken)
+    {
+        var appClient = _context.ApplicationClients.Where(x => x.Endpoint == applicationClient.Endpoint).FirstOrDefault();
```
Comment (abroodie) [inline src/SFA.DAS.PushNotifications.Functions/SFA.DAS.PushNotifications.Data/Repositories/ApplicationClientRepository.cs]: The Where filter clause can be moved to the filter in the FirstOrDefault() statement

File: src/SFA.DAS.PushNotifications.Functions/SFA.DAS.PushNotifications.Data/Repositories/ApplicationClientRepository.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            applicationClient.Status = (int)ApplicationClientStatus.Active;
+            _context.ApplicationClients.Add(applicationClient);
+        }
+        await _context.SaveChangesAsync(cancellationToken);
+        return applicationClient.Id;
+    }
+
+    public async Task RemoveWebPushNotificationSubscription(ApplicationClient applicationClient, CancellationToken cancellationToken)
+    {
+        var appClient = _context.ApplicationClients.Where(x => x.Endpoint == applicationClient.Endpoint).FirstOrDefault();
```
Comment (abroodie) [inline src/SFA.DAS.PushNotifications.Functions/SFA.DAS.PushNotifications.Data/Repositories/ApplicationClientRepository.cs]: The Where filter clause can be moved to the filter in the FirstOrDefault() statement

File: src/SFA.DAS.PushNotifications.Functions/SFA.DAS.PushNotifications.Data/Repositories/ApplicationClientRepository.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        if (appClient != null)
```
Comment (abroodie) [inline src/SFA.DAS.PushNotifications.Functions/SFA.DAS.PushNotifications.Data/Repositories/ApplicationClientRepository.cs]: invert if to reduce indentation

**Review by abroodie — CHANGES_REQUESTED** [review]
Add new SFA.DAS.PushNotifications.Application class library and then move the class SFA.DAS.PushNotifications.Functions\Services\PushNotificationsService.cs into that new project

Move classes in SFA.DAS.PushNotifications.Data.Entities namespace to SFA.DAS.PushNotifications.Model project

**Review by HussainSC — DISMISSED** [review]

**Review by alireid — COMMENTED** [review]
Cursory first glance this looks brilliant. A load of build failures, but I'm sure you are aware of that

**Review by alireid — DISMISSED** [review]

**Review by abroodie — APPROVED** [review]

**Review by HussainSC — APPROVED** [review]

**Pull request merged**